Add ob-oracle mode into bundle `oceanbase`

LuckyPickleZZ opened the pull request:
https://github.com/dbeaver/dbeaver/issues/13596
The jdbc driver jars will be fixed in the near future, our team is working on it.

LuckyPickleZZ committed:
Fix issues of @serge-rider

https://github.com/dbeaver/dbeaver/issues/13596#issuecomment-902597786
Add @2x.png.
Set 'localhost' as default host.

LuckyPickleZZ committed:
Add OB-oracle mode into bundle `ext.oceanbase`

LuckyPickleZZ committed:
Fix issues of Codacy Static Code Analysis

https://app.codacy.com/gh/dbeaver/dbeaver/pullRequest?prid=8009229
Code format.

LuckyPickleZZ committed:
Fix codacy issues

Fix codacy issues.
Modify the settings to display the system databases by default.

LuckyPickleZZ committed:
Update OceanbaseOracleSchema.java

LuckyPickleZZ committed:
Update OceanbaseOracleSchema.java

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.oceanbase/src/org/jkiss/dbeaver/ext/oceanbase/data/OceanbaseValueHandlerProvider.java:
```diff
@@ -1,25 +1,28 @@
- * limitations under the License.
- */
-
 package org.jkiss.dbeaver.ext.oceanbase.data;
```
Please return the Copyright.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.oceanbase/src/org/jkiss/dbeaver/ext/oceanbase/data/OceanbaseRowIDValueHandler.java:
```diff
@@ -0,0 +1,20 @@
+package org.jkiss.dbeaver.ext.oceanbase.data;
```
Please add the Copyright.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.oceanbase/src/org/jkiss/dbeaver/ext/oceanbase/model/plan/OceanbasePlanAnalyzer.java:
```diff
@@ -109,14 +114,12 @@ public DBPDataSource getDataSource() {
-    @NotNull
     @Override
-    public DBCPlan planQueryExecution(@NotNull DBCSession session, @NotNull String query, @NotNull DBCQueryPlannerConfiguration configuration)
+    public DBCPlan planQueryExecution(DBCSession session, String query, DBCQueryPlannerConfiguration configuration)
```
You removed all NotNull annotations in this class. Why? They are very helpful.

LuckyPickleZZ replied:
I mistakenly think this is useless... I'll fix it now and add them back.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.oceanbase/OSGI-INF/l10n/bundle.properties:
```diff
@@ -209,3 +209,369 @@ tree.users.node.name                 = \u7528\u6237
+tree.schema.node.name = \u6A21\u5F0F
+tree.tables.node.name = \u8868
+tree.table.node.name = \u8868
+tree.columns.node.name = \u5217
```
This and many parameters below are already defined above in the same file. There are a lot of duplicates rows.

LuckyPickleZZ replied:
I removed several duplications. There is basically no repetition now.

LonwoLonwo commented:
Hello @LuckyLeoZZ 
Thanks for PR. Great job!

Have you tested it? Does it work fine in Oracle mode?

No problems with "visibleIf="object.dataSource.isMySQLMode()"" plugin condition this time?

LuckyPickleZZ commented:
> Hello @LuckyLeoZZ
> Thanks for PR. Great job!
> 
> Have you tested it? Does it work fine in Oracle mode?
> 
> No problems with "visibleIf="object.dataSource.isMySQLMode()"" plugin condition this time?

You question was a good one～ In oracle mode, `isMySQLMode()` works properly. But in `OceanbaseMySQLCatlog`, first tag of `<tree>`, it will report error like `undefined variable object.dataSource`. So I have to leave this Catalogue visible even in oracle mode :(

LuckyPickleZZ committed:
Fix some review comments by @LonwoLonwo

Add @Copyrights into files.
Remove most duplications in `bundle.properties`.

serge-rider commented:
@LuckyLeoZZ English localization must be specified in bundle.properties. Chinese translation must be in bundle_zh.properties only.

LuckyPickleZZ committed:
Specify English localization

Specify English localization in bundle.properties.
https://github.com/dbeaver/dbeaver/pull/13748#issuecomment-916072598

LuckyPickleZZ commented:
> @LuckyLeoZZ English localization must be specified in bundle.properties. Chinese translation must be in bundle_zh.properties only.

I'm sorry it took me so long to modify this. Now English localization is specified in bundle.properties.

LuckyPickleZZ commented:
Excuse me, is there something wrong? Why is this PR blocked :(

LonwoLonwo commented:
Hello @LuckyLeoZZ 
You created the right bundle.properties file with English localization, but all your English words begin with a small letter :(

![2021-10-08 13_01_45-Add ob-oracle mode into bundle `oceanbase` by LuckyLeoZZ · Pull Request #13748 ·](https://user-images.githubusercontent.com/45152336/136538202-2c877f89-0c4b-4ff9-8d7c-cdaa305bdb29.png)

LuckyPickleZZ committed:
Update bundle.properties

LuckyPickleZZ commented:
> Hello @LuckyLeoZZ You created the right bundle.properties file with English localization, but all your English words begin with a small letter :(
> 
> ![2021-10-08 13_01_45-Add ob-oracle mode into bundle oceanbase by LuckyLeoZZ · Pull Request #13748 ·](https://user-images.githubusercontent.com/45152336/136538202-2c877f89-0c4b-4ff9-8d7c-cdaa305bdb29.png)

Sorry, my fault. I've fixed this.

LuckyPickleZZ commented:
There's an error in Autobuild, but I cant understand it. How can I fix this?
![image](https://user-images.githubusercontent.com/36879124/136667587-311b8ce2-b566-4451-a415-88ef16f707f8.png)

https://github.com/dbeaver/dbeaver/runs/3846687759?check_suite_focus=true

LuckyPickleZZ committed:
Merge branch 'dbeaver:devel' into devel
